### PR TITLE
Fix incorrect exception if failed to auto resolve nested dependency

### LINKED ISSUE #12: Incorrect exception thrown if failed to auto resolve nested dependency
Hi Luca,

This is an old issue I've always faced with DI52.

Consider the following scenario:

```php
class Car {
	public function __construct( Engine $e ) {}
}

class Engine {
	public function __construct( NonExisting $g ) {}
}

$container = new tad_DI52_Container();
$container->make( Car::class );
```

`( ! ) Fatal error: Uncaught RuntimeException: Error while making 'Car': 'Car' is not a bound alias or an existing class. in /shared/httpd/di52/htdocs/di52/src/tad/DI52/Container.php on line 352`

The exception tells me that `Car` doesn't exist, where actually the problem is in my `Engine` because it depends on `NonExisting`.

An explicit binding throws an accurate error:

```php
$container = new tad_DI52_Container();

$container->bind( Car::class, function ( tad_DI52_Container $c ) {
	return new Car( $c->make( NonExisting::class ) );
	// or return new Car( new NonExisting );
} );

$container->make( Car::class );
```

`( ! ) Fatal error: Uncaught RuntimeException: Error while making 'NonExisting': Class NonExisting does not exist in /shared/httpd/di52/htdocs/di52/src/tad/DI52/Container.php on line 352`

I might come up with a PR for it, but I felt like sharing this here first to give you a head start in case you want to look into this yourself.

### PR DESCRIPTION
**Pull request by Luc45**
Fixes https://github.com/lucatume/di52/issues/12

Do you see any negative side effect of this change, Luca? Seems pretty safe and straightforward to me, since an exception will be throw anyway, but I'm not 100% certain on all the variations that can throw a RuntimeException in this scenario, and whether we would like to replace the original exception with the custom message for any reason.

### CODE REVIEW

**Comment by Luc45** [comment]
Adding to the example on #12, the exception shows the whole tree of dependencies up to the point where it fails:

```php
class Car {
	public function __construct( Engine $e ) {
	}
}

class Engine {
	public function __construct( UpperEngine $u, LowerEngine $l ) {
	}
}

class UpperEngine {
	public function __construct(CombustionChamber $c) {
	}
}

class CombustionChamber {
	public function __construct(Valve $v) {
	}
}

class Valve {
	public function __construct(NonExisting1 $n) {
	}
}

class LowerEngine {
	public function __construct(DriveShaft $c) {
	}
}

class DriveShaft {
	public function __construct(Clutch $c) {
	}
}

class Clutch {
	public function __construct(NonExisting2 $c) {
	}
}
```

`RuntimeException: 'Car' => 'Engine' => 'UpperEngine' => 'CombustionChamber' => Error while making 'Valve': Class NonExisting1 does not exist`

Upon fixing `Valve`, the next binding error will show as expected:

`RuntimeException: 'Car' => 'Engine' => 'LowerEngine' => 'DriveShaft' => Error while making 'Clutch': Class NonExisting2 does not exist`

**Review by lucatume — CHANGES_REQUESTED** [review]
I would like to see some snapshot testing added for the output messages.
I can come back to this and add it if you cannot.

**Comment by Luc45** [comment]
@lucatume Once again, I find myself in a position to contribute. In the coming week(s), I should add some tests to this PR, and open a PR for https://github.com/lucatume/di52/issues/15.

**Comment by lucatume** [comment]
> @lucatume Once again, I find myself in a position to contribute. In the coming week(s), I should add some tests to this PR, and open a PR for #15.

If I'm getting to this before you do, I'm willing to add tests and include this in some PR of mine.

**Comment by Luc45** [comment]
Thanks for adding the snapshot testing, @lucatume. This is ready for a review.

For better clarity, the exception before was:

`'Car::class' is not a bound alias or an existing class.`

Now it is:

```
  'Car'
	 =>    'Engine'
	 =>    'UpperEngine'
	 =>    'CombustionChamber'
	 =>  Error while making  'Valve': Class NonExisting1 does not exist
```

File: tests/52-compat/__snapshots__/Container52CompatTest-should_throw_correct_exception_when_injecting_missing_class_in_the_constructor.snapshot-1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+  'Car'
```
Comment (lucatume) [inline tests/52-compat/__snapshots__/Container52CompatTest-should_throw_correct_exception_when_injecting_missing_class_in_the_constructor.snapshot-1]: Here and elsewhere, why all this space and misaligned format?

Ideally, I would like something more along the lines of this:
```
  Car
    => Engine
    => UpperEngine
    => CombustionChamber: Error while making  'Valve': Class NonExisting1 does not exist
```

Comment (Luc45) [reply tests/52-compat/__snapshots__/Container52CompatTest-should_throw_correct_exception_when_injecting_missing_class_in_the_constructor.snapshot-1]: Fixed in https://github.com/lucatume/di52/pull/14/commits/e623842ae44dbbcf77bd5115d14229a82ad962d4

**Comment by Luc45** [comment]
Don't merge. I'm debugging CI here.

Done. I've force-pushed it to before debugging CI to keep the git history clean.

**Review by lucatume — APPROVED** [review]

**Comment by lucatume** [comment]
And thanks again @Luc45

**Pull request merged**